src/model/Gateway/Free.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Crm\PaymentsModule\Gateways;
+
+use Crm\ApplicationModule\Config\ApplicationConfig;
+use Crm\PaymentsModule\Repository\PaymentMetaRepository;
+use Nette\Application\LinkGenerator;
+use Nette\Http\Response;
+use Nette\Localization\ITranslator;
+use Omnipay\Omnipay;
+use Omnipay\PayPal\ExpressGateway;
+use Crm\PaymentsModule\Gateways\GatewayAbstract;
+
+class Free extends GatewayAbstract
+{
+    /** @var ExpressGateway */
+    protected $gateway;
+
+    private $paymentMetaRepository;
+
+    public function __construct(
+        LinkGenerator $linkGenerator,
+        ApplicationConfig $applicationConfig,
+        Response $httpResponse,
+        PaymentMetaRepository $paymentMetaRepository,
+        ITranslator $translator
+    ) {
+        parent::__construct($linkGenerator, $applicationConfig, $httpResponse, $translator);
+        $this->paymentMetaRepository = $paymentMetaRepository;
+    }
+
+    protected function initialize()
+    {
+    }
+
+    public function begin($payment)
```

Comment:
The `begin` method cannot be empty. When I submit a sales funnel with this gateway, `begin` does nothing. There are two scenarios that can happen here:

- It can be dependent on external system which provides a redirect URL. When that's a case, CRM expects to have Omnipays `ResponseInterface` populated in `$this->response` at the end of `begin()` method. Like here: https://github.com/remp2020/crm-payments-module/blob/master/src/model/Gateway/Comfortpay.php#L57

- It can redirect the user right away (probably this scenario). Like here: https://github.com/remp2020/crm-payments-module/blob/master/src/model/Gateway/BankTransfer.php#L28

The fast scenario could be redirect user to `$this->generateReturnUrl($payment, ['vs' => $payment->variable_symbol])`. That would always confirm the payment as the processing done on the return URL checks result of `complete()` method.

Nicer implementation would be an intermediate screen, where we could select whether the payment should be successful or not so people would be able to test also error scenarios, not just sunny day scenario.

Anyway, this is a blocker and unless the payment goes correctly through the sales funnel (or anybody else calling `PaymentProcessor::begin()`, this cannot be merged.